cluster.tf

```diff
@@ -25,6 +27,22 @@ resource google_container_cluster management_cluster {
     identity_namespace = "${var.project_id}.svc.id.goog"
   }
 
+  datapath_provider = "ADVANCED_DATAPATH"
+
+  addons_config {
```

Comment:
@fristonio the question is, should these be exposed as variable or not? it would make sense, but I am no sure how to optionally set the whole block here, as these also depend on the beta provider.

Reply:
We can maybe use a [dynamic](https://www.terraform.io/docs/language/expressions/dynamic-blocks.html) block to achieve this. But I think we can also do something like enabling `dns_cache_config` and `gce_persistent_disk_csi_driver_config` always and enable config connector based on user configuration?
If want all three option to be configured by the user, exposing them using a map will be the easiest solution IMO.

Reply:
Ah, good to know there is dynamic block now, that's quite neat! On second thought I am still not sure how to define the logic around `provider = google-beta` based on that... maybe it's not worth the trouble given this is a rather single-purpose module.